### PR TITLE
Allow for continue in while loops

### LINKED ISSUE #4810: Continue for while loops
### Suggestion

Adding `continue` to work with while loops, currently `continue can only be used in loops` is what the error states.

### Why?

Because most programming languages have continue allowed in while loops and I thought Skript already supported it.

### PR DESCRIPTION
**Pull request by kiip1**
### Description
Allows for `continue` to be used in while loops.

---
**Target Minecraft Versions:** All
**Requirements:** None
**Related Issues:**
+ #4810

### CODE REVIEW

**Comment by AyhamAl-Ali** [comment]
Thanks for this nice PR \:D
I think it should probably wait for #4595 so it can benifit from `LoopSection`

**Comment by kiip1** [comment]
`LoopSection` surely would be useful when merged, but no need to wait on it as only 3 lines of code will change.

**Comment by AyhamAl-Ali** [comment]
If that PR is going to take some time after 2.7 then yea no need to wait that long but hopefully they will be merged together.

**Comment by TheLimeGlass** [comment]
Converted to draft as this pull will need updating when the Structure API gets merged. https://github.com/SkriptLang/Skript/pull/4108

File: src/main/java/ch/njol/skript/effects/EffContinue.java [hunk 1]
```diff
@@ -48,16 +52,20 @@ public class EffContinue extends Effect {
...
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		List<SecLoop> loops = getParser().getCurrentSections(SecLoop.class);
-		if (loops.isEmpty()) {
+		List<TriggerSection> currentSections = ParserInstance.get().getCurrentSections().stream()
+			.filter(s -> s instanceof SecLoop || s instanceof SecWhile)
+			.collect(Collectors.toList());
+		
+		if (currentSections.isEmpty()) {
 			Skript.error("Continue may only be used in loops");
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/effects/EffContinue.java:64]: The error should also reflect while loops.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
Looks good, just an update to the error message so users aren't misinformed about while loops aswell.

**Review by TheLimeGlass — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]
LGTM STICKER

**Pull request merged**